components/script/dom/request.rs

```diff
@@ -308,21 +308,26 @@ impl Request {
         // Step 27
         let mut headers_copy = r.Headers();
 
-        // This is equivalent to the specification's concept of
-        // "associated headers list".
-        if let RequestInfo::Request(ref input_request) = input {
-            headers_copy = input_request.Headers();
-        }
-
         // Step 28
```

Comment:
The step numbers seem to have changed (though that's probably my fault!), feel free to update them if you feel like it :).